gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -609,29 +653,37 @@ private int peekKeyword() throws IOException {
     String keyword;
     String keywordUpper;
     int peeking;
-    if (c == 't' || c == 'T') {
```

Comment:
FWIW, `Character.toLowerCase(char)` does not depend on the locale and the only characters that map to `t` are `t` and `T`. Same story for `f` and `n`. We might feel a bit icky about it, but it's not wrong.

Reply:
Yes you are right, I wasn't very precise with my message. For `keyword.toUpperCase()` it would definitely be necessary to specify a `Locale` though.

> and the only characters that map to `t` are `t` and `T`. Same story for `f` and `n`

I would add here "... at the moment" though. The JDK is regularly updated with new Unicode data, and there is for example already the Kelvin sign character `U+212A` for which `Character.toLowerCase('\u212A') == 'k'` is true. So it is not completely unrealistic that a future Unicode version includes a character whose lowercase variant is `t`, `f` or `n`.

Either way, I have adjusted the code in https://github.com/marten-voorberg/DD2480-gson/pull/49 (mentioned below) to be close to the original code again and to not perform any Unicode case conversions anymore (I assume that would also be more performant, but I have not verified it).